os/arch/arm/src/imxrt/imxrt_pit.c

```diff
@@ -151,3 +216,381 @@ uint64_t imxrt_pit_getlifetimetimercount(PIT_Type *base)
+
+	status->flags = 0;
+	if (priv->started) {
+		status->flags |= TCFLAGS_ACTIVE;
+	}
+
+	if (priv->callback) {
+		status->flags |= TCFLAGS_HANDLER;
+	}
+
+	/* Return the actual timeout in microseconds */
+	timeout_ticks = imxrt_pit_gettimerperiod(priv->pit->base, priv->channel);
+	status->timeout = ((uint64_t)USEC_PER_SEC * (uint64_t)timeout_ticks) / freq;
+
+	/* Get the time remaining until the timer expires (in microseconds). */
+	cur_ticks = imxrt_pit_getcurrenttimercount(priv->pit->base, priv->channel);
+	ticks = timeout_ticks - cur_ticks;
+	status->timeleft = ((uint64_t)USEC_PER_SEC * (uint64_t)ticks) / freq;
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_setcallback
+ *
+ * Description:
+ *   Call this user provided timeout callback.
+ *
+ * Input Parameters:
+ *   lower    - A pointer the publicly visible representation of the
+ *              "lower-half" driver state structure.
+ *   callback - The new timer expiration function pointer. If this
+ *              function pointer is NULL, then the reset-on-expiration
+ *              behavior is restored,
+ *  arg       - Argument that will be provided in the callback
+ *
+ * Returned Values:
+ *   The previous timer expiration function pointer or NULL is there was
+ *   no previous function pointer.
+ *
+ ****************************************************************************/
+static void imxrt_pit_setcallback(struct timer_lowerhalf_s *lower,
+								tccb_t callback, void *arg)
+{
+	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)lower;
+	irqstate_t flags = irqsave();
+
+	/* Save the new callback */
+	priv->callback = callback;
+	priv->arg      = arg;
+
+	if (priv->callback != NULL) {
+		imxrt_pit_setisr(priv->pit, imxrt_pit_handler, priv);
+		imxrt_pit_enableinterrupts(priv->pit->base, priv->channel, kPIT_TimerInterruptEnable);
+	} else {
+		imxrt_pit_disableinterrupts(priv->pit->base, priv->channel, kPIT_TimerInterruptEnable);
+		imxrt_pit_setisr(priv->pit, NULL, NULL);
+	}
+
+	irqrestore(flags);
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_ioctl
+ *
+ * Description:
+ *   Any ioctl commands that are not recognized by the "upper-half" driver
+ *   are forwarded to the lower half driver through this method.
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *   cmd   - The ioctl command value
+ *   arg   - The optional argument that accompanies the 'cmd'.  The
+ *           interpretation of this argument depends on the particular
+ *           command.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int imxrt_pit_ioctl(struct timer_lowerhalf_s *lower, int cmd,
+						unsigned long arg)
+{
+	return OK;

```

Comment:
-ENOTTY

Reply:
Implemented ioctl for lifetimer. No more required to return -ENOTTY